bep_webrtc.rst

```diff
@@ -0,0 +1,239 @@
+:BEP: XXX
+:Title: Websocket tracker protocol for WebRTC transport
+:Version: $Revision$
+:Last-Modified: $Date$
+:Author:  Yoann Ciabaud <y.ciabaud@gmail.com>
+:Status:  Draft
+:Type:    Standards Track
+:Content-Type: text/x-rst
+:Created: 02-Aug-2016
+:Post-History:
+
+
+Abstract
+========
+Webapps are more and more used to publish and consume content over the
+internet, Bittorrent could be used to optimize the server and network resources
+needed to distribute files but the need for a client software prevents its
+usage in a browser environment.
+
+This BEP documents how to add support for WebRTC signalling on a tracker in
+order to make the webapps able to connect peers using the WebRTC API [1].
+Once a communication channel is open between the peers using WebRTC, the
+standard Bittorent peer connection protocol is used.
+
+Due to the WebRTC signalling process, the tracker needs to push messages to the
+clients with a full-duplex communication channel available in a web browser.
+At the moment of this BEP, Websockets [2] are the only compatible standard
+technology.
+
+WebRTC signalling process
+=========================
+The diagram below explains the messages involved in the WebRTC signalling
+process.
+
+::
+
+  peer 1                 tracker                  peer 2
+
+    |                       |                       |
+    |                       |                       |
+    | announce start        |                       |
+    |   numwant=x           |                       |
+    |   x SDP offers        |                       |
+    | >-------------------> |                       |
+    |                       |                       |
+    |                       |  SDP offer            |
+    |                       | >-------------------> |
+    |                       |                       |
+    |                       |  SDP answer           |
+    |                       | <-------------------< |
+    |                       |                       |
+    |  SDP answer           |                       |
+    | <-------------------< |                       |
+    |                       |                       | connection established
+   .|.. .. .. .. .. .. .. .. .. .. .. .. .. .. .. ..|.. .
+    |                                               |
+    V                                               V
+
+Websocket tracker protocol
+==========================
+
+credits
+-------
+The websocket tracker protocol has been designed for the Webtorrent [3] project
+by Feross Aboukhadijeh and hundreds of open source contributors.
+
+overview
+--------
+The websocket tracker uses JSON payloads reflecting the HTTP request parameters
+and an additionnal action property used to switch between announce and other
+actions (ex. scrape). If the announce URL of the torrent contains the ws or wss
```

Comment:
I STRONGLY recommend that we formally drop support for insecure websockets. It's almost 2017, there is no reason to introduce a potential security vulnerability or vector for surveillance. Especially with Let's Encrypt available for free, there is simply no excuse for running an insecure public service. WebRTC itself already disallows insecure connections, we should incorporate that momentum into this BEP.

I suggest we change this explicitly disallow 'ws' and only allow 'wss' here.